lib/buffer.js

```diff
@@ -371,7 +371,7 @@ Buffer.prototype.toString = function() {
     var result = slowToString.apply(this, arguments);
   }
   if (result === undefined)
-    throw new Error('toString failed');
+    throw new Error('toString failed - buffer larger than maximum string size ' + binding.kStringMaxLength);
```

Comment:
After the number I would put "bytes". It may be confusing whether max size is string characters or not.